src/closure.js

```diff
@@ -5,17 +5,41 @@ const counter = () => {
   // Example: const newCounter = counter();
   // newCounter(); // 1
   // newCounter(); // 2
+  let counterItem = 0;
+  return () => {
+    counterItem++;
+    return counterItem;
+  };
 };
 
 const counterFactory = () => {
   // Return an object that has two methods called `increment` and `decrement`.
   // `increment` should increment a counter variable in closure scope and return it.
   // `decrement` should decrement the counter variable and return it.
+  let counterItem = 0;
+  return {
+    increment: () => {
+      counterItem++;
+      return counterItem;
+    },
+    decrement: () => {
+      counterItem--;
+      return counterItem;
+    }
+  };
 };
 
 const limitFunctionCallCount = (cb, n) => {
   // Should return a function that invokes `cb`.
   // The returned function should only allow `cb` to be invoked `n` times.
+  let counterNumber3 = 1;
+  const invocationLimit = n;
+  return (...args) => {
+    if (counterNumber3 <= invocationLimit) {
```

Comment:
Usually it's better to put the smaller statement in the `if` statement when there is a `return`.  So do this:
```
if (counterNumber3 > invocationLimit) return null;
// everything else...
```